pkg/provider/nomad/nomad_test.go

```diff
@@ -161,70 +300,144 @@ func Test_getNomadServiceData(t *testing.T) {
+	require.NoError(t, err)
+
+	// make the query for services
+	items, err := p.getNomadServiceData(context.TODO())
+	require.NoError(t, err)
+	require.Len(t, items, 2)
+}
+
+func Test_getNomadServiceData_TCP(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch {
+		case strings.HasSuffix(r.RequestURI, "/v1/jobs"):
+			_, _ = w.Write(responses["jobs_job7"])
+		case strings.HasSuffix(r.RequestURI, "/v1/job/job7"):
+			_, _ = w.Write(responses["job_job7_TCP"])
+		case strings.HasSuffix(r.RequestURI, "/v1/service/job7"):
+			_, _ = w.Write(responses["service_job7"])
+		}
+	}))
+
+	t.Cleanup(ts.Close)
+
+	p := new(Provider)
+	p.SetDefaults()
+	p.Endpoint.Address = ts.URL
+	err := p.Init()
+	require.NoError(t, err)
+
+	// fudge client, avoid starting up via Provide
+	p.client, err = createClient(p.namespace, p.Endpoint)
+	require.NoError(t, err)
+
+	// make the query for services
+	items, err := p.getNomadServiceData(context.TODO())
+	require.NoError(t, err)
+	require.Len(t, items, 1)
+}
+
+func Test_getNomadServiceData_UDP(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch {
+		case strings.HasSuffix(r.RequestURI, "/v1/jobs"):
+			_, _ = w.Write(responses["jobs_job8"])
+		case strings.HasSuffix(r.RequestURI, "/v1/job/job8"):
+			_, _ = w.Write(responses["job_job8_UDP"])
+		case strings.HasSuffix(r.RequestURI, "/v1/service/job8"):
+			_, _ = w.Write(responses["service_job8"])
+		}
+	}))
+
+	t.Cleanup(ts.Close)
+
+	p := new(Provider)
+	p.SetDefaults()
+	p.Endpoint.Address = ts.URL
+	err := p.Init()
+	require.NoError(t, err)
+
+	// fudge client, avoid starting up via Provide
+	p.client, err = createClient(p.namespace, p.Endpoint)
+	require.NoError(t, err)
+
+	// make the query for services
+	items, err := p.getNomadServiceData(context.TODO())
+	require.NoError(t, err)
+	require.Len(t, items, 1)
+}
+
+func Test_getNomadServiceData_ScalingEnabled_Stopped(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch {
+		case strings.HasSuffix(r.RequestURI, "/v1/jobs"):
+			_, _ = w.Write(responses["jobs_job9"])
+		case strings.HasSuffix(r.RequestURI, "/v1/job/job9"):
+			_, _ = w.Write(responses["job_job9_ScalingEnabled_Stopped"])
+		case strings.HasSuffix(r.RequestURI, "/v1/service/job9"):
+			_, _ = w.Write(responses["service_job9"])
+		}
+	}))
+
+	t.Cleanup(ts.Close)
+
+	p := new(Provider)
+	p.SetDefaults()
+	p.Endpoint.Address = ts.URL
+	err := p.Init()
+	require.NoError(t, err)
+
+	// fudge client, avoid starting up via Provide
+	p.client, err = createClient(p.namespace, p.Endpoint)
+	require.NoError(t, err)
+
+	// make the query for services
+	items, err := p.getNomadServiceData(context.TODO())
+	require.NoError(t, err)
+
+	// Should not be listed as job is stopped
+	require.Empty(t, items)
+}
+
+func setup() error {
+	responsesDir := "./nomad_responses"
```

Comment:
```suggestion
	responsesDir := "./fixtures"
```

Could you please rename `nomad_responses` directory to `fixtures`?

Reply:
changes have been made